Makefile

```diff
@@ -29,15 +29,48 @@ SHELL := /usr/bin/env bash
 .SUFFIXES:
 FORCE:
 
+HOST_OS ?= $(shell uname -s | tr A-Z a-z)
+HOST_ARCH ?= $(shell uname -m)
+ifeq (x86_64, $(HOST_ARCH))
+	HOST_ARCH = amd64
+endif
+
 bin_dir := _bin
 
-$(bin_dir):
+$(bin_dir) $(bin_dir)/scratch:
 	mkdir -p $@
 
-include modules/tools/00_mod.mk
+include modules/**/00_mod.mk
+
+.PHONY: images-learn-sha
+images-learn-sha: $(bin_dir) | $(NEEDS_CRANE)
+	rm -rf ./$(bin_dir)/downloaded/images/
+	mkdir -p ./$(bin_dir)/scratch/
+	
+	$(eval export LEARN_FILE=$(CURDIR)/$(bin_dir)/scratch/learn_images_file)
+	echo -n "" > "$(LEARN_FILE)"
+
+	for image in $(images_amd64); do \
+		image_no_digest=$$(echo -n "$${image}" | cut -d@ -f1); \
+		find=$$(echo -n "$${image}" | cut -d@ -f2); \
+		replace=$$($(CRANE) digest --platform "linux/amd64" "$${image_no_digest}"); \
+		echo "s|$${find}|$${replace}|g" >> "$(LEARN_FILE)"; \
+	done
+
+	for image in $(images_arm64); do \
+		image_no_digest=$$(echo -n "$${image}" | cut -d@ -f1); \
+		find=$$(echo -n "$${image}" | cut -d@ -f2); \
+		replace=$$($(CRANE) digest --platform "linux/arm64" "$${image_no_digest}"); \
+		echo "s|$${find}|$${replace}|g" >> "$(LEARN_FILE)"; \
+	done
+
+	while read p; do \
+		find ./modules/ -maxdepth 2 -name "00_mod.mk" -type f -exec sed -i "$$p" {} \;; \
+	done <"$(LEARN_FILE)"
```

Comment:
suggestion: To me this is complicated enough that it would benefit from being a standalone bash script which `make images-learn-sha` simply calls.

It'd be easier to read (fewer `\`s and `;`s) and debug (e.g. adding shellcheck later).

Plus this make target is phony but modifies named files. That definitely feels like more of a bash script behaviour to me. What do you think?

Reply:
I moved the logic to an bash script.